test/al_aws_collector_test.js

```diff
@@ -524,6 +536,56 @@ describe('al_aws_collector tests', function() {
         });
     });
 
+    describe('checkin error throttling', function () {
+        var checkinContext = {
+            invokedFunctionArn: colMock.FUNCTION_ARN,
+            functionName: colMock.FUNCTION_NAME
+        };
+        const stub = sinon.stub();
+        let describeCallCount = 0;
+        const MAX_RETRIES = 3;
+        before(function () {
+            stub.onCall(0).returns(colMock.CF_DESCRIBE_STACKS_FAILED_THROTTLING_ERROR);
+            stub.onCall(1).returns(colMock.CF_DESCRIBE_STACKS_FAILED_THROTTLING_ERROR);
+            stub.onCall(2).returns(colMock.CF_DESCRIBE_STACKS_FAILED_THROTTLING_ERROR);
+            stub.onCall(3).returns(colMock.CF_DESCRIBE_STACKS_FAILED_THROTTLING_ERROR);
+            AWS.mock('CloudFormation', 'describeStacks', function (params, callback) {
+                if (describeCallCount < MAX_RETRIES) {
+                    describeCallCount++;
```

Comment:
maybe get rid of `describeCallCount` and use 
`stub.onCall(4).returns(null,colMock.CF_DESCRIBE_STACKS_RESPONSE)`